File: app/src/main/java/com/stathis/elmepaunivapp/models/Lesson.java
package com.stathis.elmepaunivapp.models;

public class Lesson {

    private String name;
    private String mandatory;
    private String description;
    private String direction;

    public Lesson(String name, String mandatory, String description, String direction) {
        this.name = name;
        this.mandatory = mandatory;
        this.description = description;
        this.direction = direction;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getMandatory() {
        return mandatory;
    }

    public void setMandatory(String mandatory) {
        this.mandatory = mandatory;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getDirection() {
        return direction;
    }

    public void setDirection(String direction) {
        this.direction = direction;
    }
}

File: app/src/main/java/com/stathis/elmepaunivapp/ui/syllabus_lessons/model/Lesson.kt
package com.stathis.elmepaunivapp.ui.syllabus_lessons.model

import android.os.Parcelable
import com.stathis.elmepaunivapp.abstraction.LocalModel
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Lesson(

    var name: String,
    val mandatory: String,
    val description: String,
    val direction: String

) : LocalModel, Parcelable {
    override fun equalsContent(obj: LocalModel): Boolean = when(obj){
        is Lesson -> name == obj.name && mandatory == obj.mandatory && description == obj.description && direction == obj.direction
        else -> false
    }
}
